runners/core-java/src/main/java/org/apache/beam/runners/core/PushbackSideInputDoFnRunner.java

```diff
@@ -79,14 +78,14 @@ public void startBundle() {
     for (WindowedValue<InputT> windowElem : elem.explodeWindows()) {
       BoundedWindow mainInputWindow = Iterables.getOnlyElement(windowElem.getWindows());
       boolean isReady = !notReadyWindows.contains(mainInputWindow);
-      for (PCollectionView<?> view : views) {
-        BoundedWindow sideInputWindow =
-            view.getWindowingStrategyInternal()
-                .getWindowFn()
-                .getSideInputWindow(mainInputWindow);
-        if (!sideInputReader.isReady(view, sideInputWindow)) {
-          isReady = false;
-          break;
+      if (isReady) {
```

Comment:
nit: Curly nesting is getting a bit much. Maybe break out the `isReady` as a little helper method, then this short-circuit and the `break` can both be `return`, which is nice to read.